### PR TITLE
Feature - Extracting modals to their own files in console

### PR DESCRIPTION
**Pull request by MoiyadKayda**
## Description
I have also changed components to use ui-react, wherever possible.

## Screenshots:
image
image
image
image
image

## Checklist:

- [x] I have updated the CHANGELOG with a summary of my changes

### CODE REVIEW

File: src/packages/console/src/components/newPlanDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+                <br />
+                <Text>Since you haven&apos; got a current subscription for this account, we&apos;ll need your credit card details to continue.</Text>
+                <TextField
+                  variant='outlined'
+                  margin='normal'
+                  fullWidth
+                  label='Card Details'
+                  disabled={isLoading}
+                  InputLabelProps={{ shrink: true }}
+                  InputProps={{
```
Comment (MoiyadKayda) [inline src/packages/console/src/components/newPlanDialog.tsx]: How to convert this to use ui-react

File: src/packages/console/src/components/newPlanDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+const StripeInput = (props: StripeInputProps): React.ReactElement => {
+  const elementRef = React.useRef();
+  React.useImperativeHandle(props.inputRef, () => ({
+    focus: () => elementRef.current.focus,
+  }));
+  return (
+    <props.component
+      // eslint-disable-next-line no-return-assign
+      onReady={(element) => (elementRef.current = element)}
+      {...props}
```
Comment (MoiyadKayda) [inline src/packages/console/src/components/newPlanDialog.tsx:137]: Should I make it a separate component

File: src/packages/console/src/components/newPlanDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+interface StripeInputProps {
+  component: React.Component;
+  inputRef: React.Ref<HTMLInputElement>;
+}
+
+const StripeInput = (props: StripeInputProps): React.ReactElement => {
+  const elementRef = React.useRef();
+  React.useImperativeHandle(props.inputRef, () => ({
+    focus: () => elementRef.current.focus,
```
Comment (MoiyadKayda) [inline src/packages/console/src/components/newPlanDialog.tsx:131]: there also some problem with this line, datatype issue here

**Comment by MoiyadKayda** [comment]
Please check the output of the newPlan dialog, I was unable to open it as it requires some details from your backend.

File: src/packages/console/src/pages/canvasPage.tsx [hunk 4]
```diff
@@ -140,25 +136,7 @@ export const CanvasPage = (): React.ReactElement => {
...
-        <DialogActions>
-          <Button onClick={onStartOverAlertConfirmClicked} color='secondary' autoFocus>
-            Start over anyway
-          </Button>
-          <Button onClick={onStartOverAlertCloseClicked} color='primary'>
-            Cancel
-          </Button>
-        </DialogActions>
-      </Dialog>
+      <StartOverDialog isOpen={isShowingStartOverAlert} onCloseClicked={onStartOverAlertCloseClicked} onStartOverAlertConfirmClicked={onStartOverAlertConfirmClicked} />
```
Comment (krishan711) [inline src/packages/console/src/pages/canvasPage.tsx]:
```suggestion
      <StartOverDialog 
        isOpen={isShowingStartOverAlert} 
        onCloseClicked={onStartOverAlertCloseClicked} 
        onStartOverAlertConfirmClicked={onStartOverAlertConfirmClicked}
      />
```

File: src/packages/console/src/pages/accountPage.tsx [hunk 5]
```diff
@@ -330,119 +317,9 @@ export const AccountPage = (props: IAccountPageProps): React.ReactElement => {
...
-                        Upgrade
-                      </Button>
-                    )}
-                  </DialogActions>
-                )}
-              </ElementsConsumer>
-            )}
-          </Dialog>
-        )}
+        {newPlan && <NewPlanDialog isUpgradeDialogLoading={isUpgradeDialogLoading} newPlan={newPlan} currentPlan={currentPlan} upgradeCardError={upgradeCardError} upgradeDiscountCode={upgradeDiscountCode} upgradeDiscountCodeError={upgradeDiscountCodeError} onUpgradeDialogClosed={onUpgradeDialogClosed} onUpgradeDiscountCodeChanged={onUpgradeDiscountCodeChanged} upgradeError={upgradeError} onUpgradeDialogUpgradeClicked={onUpgradeDialogUpgradeClicked} />}
```
Comment (krishan711) [inline src/packages/console/src/pages/accountPage.tsx]:
```suggestion
        {newPlan && (
          <NewPlanDialog 
            isUpgradeDialogLoading={isUpgradeDialogLoading} 
            newPlan={newPlan} 
            currentPlan={currentPlan} 
            upgradeCardError={upgradeCardError} 
            upgradeDiscountCode={upgradeDiscountCode}
             upgradeDiscountCodeError={upgradeDiscountCodeError} 
             onUpgradeDialogClosed={onUpgradeDialogClosed} 
             onUpgradeDiscountCodeChanged={onUpgradeDiscountCodeChanged}
             upgradeError={upgradeError} 
             onUpgradeDialogUpgradeClicked={onUpgradeDialogUpgradeClicked} 
           />
         )}
```

File: src/packages/console/src/components/newPlanDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+  },
+}));
+
+export interface INewPlanDialogProps {
+  // isOpen: boolean;
+  isUpgradeDialogLoading: boolean;
+  newPlan: IPlan;
+  currentPlan: IPlan;
+  upgradeCardError: string;
+  upgradeDiscountCode: string;
```
Comment (krishan711) [inline src/packages/console/src/components/newPlanDialog.tsx:17]: this state doesnt need to be passed it, it can be managed in the dialog

Comment (krishan711) [reply src/packages/console/src/components/newPlanDialog.tsx:17]: let me know if this makes sense, if it doesnt i'll take over once all the modals are moved into their own files

Comment (MoiyadKayda) [reply src/packages/console/src/components/newPlanDialog.tsx:17]: upgradeCardError, upgradeDiscountCode , these can be managed

**Review by krishan711 — APPROVED** [review]

**Pull request merged**